pandas/tests/reductions/test_reductions.py

```diff
@@ -688,6 +688,12 @@ def test_ops_consistency_on_empty(self, method):
         result = getattr(Series(dtype=float), method)()
         assert pd.isna(result)
 
+        # Empty Mean
+        if method == "mean":
```

Comment:
I think this is already being tested directly above? Probably want tests for non-empty input also (i.e., non-empty with some missing, non-empty with no missing, non-empty with all missing, parametrizing over skipna).

Reply:
Yes redundant now. Previously `np.nan` was being returned for empty values.

Reply:
Should i add those tests [here](https://github.com/pandas-dev/pandas/blob/5fdd6f50ac5ac2de939ef5b849e4c41e27e623a3/pandas/tests/reductions/test_reductions.py#L565)?

Reply:
These cases would I hope already be tested for Series but maybe doesn't hurt to add some more tests under `/pandas/tests/series/test_arithmetic.py` (they could always be deduplicated later if already tested) since I believe masked arrays boxed inside Series hit this path. You could test means of both nullable integer and boolean dtypes.

As a follow-up enhancement would also be nice to use this to directly implement mean on masked arrays:

```python
[ins] In [1]: import pandas as pd

[ins] In [2]: arr = pd.array([1, 2, 3])

[ins] In [3]: arr.sum()
Out[3]: 6

[ins] In [4]: arr.mean()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-057c7202e3e0> in <module>
----> 1 arr.mean()

AttributeError: 'IntegerArray' object has no attribute 'mean'
```

Reply:
> These cases would I hope already be tested for Series but maybe doesn't hurt to add some more tests under `/pandas/tests/series/test_arithmetic.py` (they could always be deduplicated later if already tested) since I believe masked arrays boxed inside Series hit this path. You could test means of both nullable integer and boolean dtypes.

I'm not sure  exactly where to add the tests? In `/pandas/tests/series/test_arithmetic.py` should I add them in [Unsorted](https://github.com/pandas-dev/pandas/blob/18b4864b7d8638c3101ec11fd6562d2fbfe872a8/pandas/tests/series/test_arithmetic.py#L640) section for now?
Also, I would be adding tests for: 
1. Nullable integer dtype
2. Nullable boolean dtype
yes?
and how do I check if the tests are passing?

> As a follow-up enhancement would also be nice to use this to directly implement mean on masked arrays:
> 
> ```python
> [ins] In [1]: import pandas as pd
> 
> [ins] In [2]: arr = pd.array([1, 2, 3])
> 
> [ins] In [3]: arr.sum()
> Out[3]: 6
> 
> [ins] In [4]: arr.mean()
> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> <ipython-input-4-057c7202e3e0> in <module>
> ----> 1 arr.mean()
> 
> AttributeError: 'IntegerArray' object has no attribute 'mean'
> ```

yes that makes sense, although I'm not sure how to go about doing this.

Reply:
@jorisvandenbossche had suggested putting some tests here which seems easiest (maybe add a couple assertions regarding empty or all-NA Series with a nullable type):

https://github.com/pandas-dev/pandas/blob/5fdd6f50ac5ac2de939ef5b849e4c41e27e623a3/pandas/tests/reductions/test_reductions.py#L683

so you can ignore my suggestion of putting them elsewhere. (No need to worry about the follow-up here, that can always be handled in a separate PR.)

Also just as a pointer, it's a good idea not to touch the master branch of your fork except when merging with upstream master and only make changes on a separate branch (from personal experience it can create headaches / confusion later).

Reply:
> @jorisvandenbossche had suggested putting some tests here which seems easiest (maybe add a couple assertions regarding empty or all-NA Series with a nullable type):
> 
> https://github.com/pandas-dev/pandas/blob/5fdd6f50ac5ac2de939ef5b849e4c41e27e623a3/pandas/tests/reductions/test_reductions.py#L683

Okay, i'll add some tests here.
 
> Also just as a pointer, it's a good idea not to touch the master branch of your fork except when merging with upstream master and only make changes on a separate branch (from personal experience it can create headaches / confusion later).

Thanks for the pointer! I will work on a separate branch from now 👍

Reply:
Also probably good to merge master since it's been a while

Reply:
I have added these tests under https://github.com/pandas-dev/pandas/blob/5fdd6f50ac5ac2de939ef5b849e4c41e27e623a3/pandas/tests/reductions/test_reductions.py#L683
![image](https://user-images.githubusercontent.com/32221186/97133491-41a46180-1770-11eb-98f5-5cbc7013fcf3.png)

Any quick fixes? also, how do I ensure these tests are passing?

Reply:
Hi @dsaxton, 
pushed to branch named `tests`.  Please have a look

Reply:
@Akshatt please push it to this branch (the same as this PR is based on, so which is your master branch), then we can see the tests here in the PR

Reply:
Yes, sorry @Akshatt if that was confusing. I meant _in the future_ it's good to avoid modifying your master branch, not to work on a separate branch for this PR.

Regarding your test you'll also want to explicitly assert `result is pd.NA` which (oddly enough) is slightly different from `pd.isna(result)`.